__tests__/github.test.js

```diff
@@ -0,0 +1,55 @@
+const pool = require('../lib/utils/pool');
+const setup = require('../data/setup');
+const request = require('supertest');
+const app = require('../lib/app');
+const GithubUser = require('../lib/models/GithubUser');
+
+jest.mock('../lib/utils/github');
+
+describe('backend-gitty routes', () => {
+  beforeEach(() => {
+    return setup(pool);
+  });
+
+  afterAll(() => {
+    pool.end();
+  });
+
+  it('should redirect to the github oauth page upon login', async () => {
+    const req = await request(app).get('/api/v1/github/login');
+
+    expect(req.header.location).toMatch(
+      /https:\/\/github.com\/login\/oauth\/authorize\?client_id=[\w\d]+&scope=user&redirect_uri=http:\/\/localhost:7890\/api\/v1\/github\/login\/callback/i
+    );
+  });
+
+  it('should login and redirect users to /api/v1/github/posts', async () => {
+    const req = await request
+      .agent(app)
+      .get('/api/v1/github/login/callback?code=42')
+      .redirects(1);
+
+    expect(req.body).toEqual({
```

Comment:
the expect here should verify the redirect path, or expecting to see posts (not a user)